agent/agent.go

```diff
@@ -1396,6 +1396,8 @@ func (a *Agent) consulConfig() (*consul.Config, error) {
 
 	base.ConfigEntryBootstrap = a.config.ConfigEntryBootstrap
 
+	base.RaftApplyTimeout = a.config.RaftApplyTimeout
```

Comment:
I just noticed that it has to be checked if the Timeout is `0` because otherwise that would override the default.

Reply:
Ahh thanks for pointing out! I'll update `DefaultSource()` for this problem.

Reply:
Resolved in the latest master, could you take a look?